### ISSUE TITLE
[class-name-casing] apply same rules on type alias

### ISSUE THREAD

**Issue by Hotell** [typescript-eslint/typescript-eslint#671]
Currently `class-name-casing` applies only for `class` and `interface`.  It would be nice ( for consistent codebase ) to apply this rule also on `type`. probably `enum` should use this as well.

Maybe it should be extracted to separate rule as `class-name-casing` is too confusing by it's name.
Suggested rule rename/new name (for interface/type/enum) -> `types-casing`

**Expected Result**

```ts
// LINT ERROR
type user = {
  name: string
  age: number
}
```

**Actual Result**

```ts
// NO LINT ERROR 🥺
type user = {
  name: string
  age: number
}
```

**Comment by j-f1** [comment]
`symbol-name-casing` or something similar since that’s the term for all these things?

**Comment by Hotell** [comment]
> `symbol-name-casing` or something similar since that’s the term for all these things?
. `
hmm symbol can mean various things right? in JS terminology -> `Symbol` primitive type, in TS terminology -> inert  entities => types 

I'm not picky on name, rather than having the described functionality 🙏
